libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -477,9 +498,20 @@ internal class MapboxTripSession(
      *
      * @return an initialized [NavigationStatus] if no errors, invalid otherwise
      */
-    override fun updateLegIndex(legIndex: Int): Boolean {
-        invalidateLatestBannerInstructionEvent()
-        return navigator.updateLegIndex(legIndex)
+    override fun updateLegIndex(legIndex: Int, callback: LegIndexUpdatedCallback) {
+        updateLegIndexCallback = callback
```

Comment:
I suggest using `try..finally` to make sure the callback is always triggered. 
This way we won't need to store it in a property.

Reply:
great idea 👍